### ISSUE TITLE
Nonsensical slice bounds seem to always be accepted

### ISSUE THREAD

**Issue by refi64** [astral-sh/ty#3547]
### Summary

([playground](https://play.ty.dev/441d055c-d96c-4fb1-9e1e-94963981f785)):

```python
# These all pass:
from typing import Iterable
def a(s: str) -> str: return s[int:[]:...]
def b(s: list[int]) -> list[int]: return s[:set()]
# Normal indexing fails as expected:
def c(s: str) -> str: return s[int]
```

Basically any variant of slicing that I can imagine passes type checking, no matter the values given, across both strings and lists in the very least.

### Version

0f3ea5adc (playground)

**Comment by MeGaGiGaGon** [comment]
This is the same conclusion I came to in https://github.com/astral-sh/ty/issues/2595#issuecomment-3791049391. By using `TypeOf` (plus a function to avoid a `@TODO`), you can see that any non-well behaved slice becomes `Slice[Any, Any, Any]`:
https://play.ty.dev/b7220037-89c9-40c5-9d17-029efc5aa3fc
```py
from ty_extensions import TypeOf

def foo(x: TypeOf[int:[]:...]):
    reveal_type(x)  # Revealed type: `slice[Any, Any, Any]`
```

**Comment by carljm** [comment]
Yeah we should have tighter validation here, thanks for the report!

**Comment by jelle-openai** [comment]
Here is a slightly more realistic example where ty produces a false negative:

```python
def chars_at(s: str, start: float, end: float):
    return s[start:end]
```

Most other type checkers flag this (string only supports int slices, not float slices), but ty doesn't.
